### PR TITLE
feat(textfield): add textfield component

### PR DESCRIPTION
**Pull request by kduret**
+ introduce clsx, which is lightweight compared to classnames

image

### CODE REVIEW

File: src/TextField/index.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+) {
+  const classes = useStyles();
+
+  return (
+    <MuiTextField
+      InputProps={{
+        ...InputProps,
+        classes: {
+          ...InputProps.classes,
+          root: clsx(
```
Comment (bdauria) [inline src/TextField/index.jsx]: Not entirely sure why we need clsx. Mui already supports class composition, right?

File: src/TextField/index.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  },
+  inputHelperText: {
+    marginTop: 4,
+    textAlign: 'right',
+  },
+}));
+
+const TextField = forwardRef(function TextField(
+  { InputProps, InputLabelProps, FormHelperTextProps, ...other },
+  ref,
```
Comment (bdauria) [inline src/TextField/index.jsx]: For what use case do you need the initial ref?

Comment (kduret) [reply src/TextField/index.jsx]: one of the use case is to manage focus

File: src/TextField/index.jsx (hunk 1 above)
Comment (kduret) [reply src/TextField/index.jsx]: If I instantiate centreon-ui/TextField and I want to override "input" class, I need to use clsx to keep properties of this component (L12).
Do you know an other way to do it ?

Comment (kduret) [reply src/TextField/index.jsx]: If i override only root, I can probably just use className prop, but in the case of nested class, I need to use this way

**Pull request merged**